glean-core/rlb/src/lib.rs

```diff
@@ -379,6 +376,13 @@ pub fn shutdown() {
     if let Err(e) = dispatcher::shutdown() {
         log::error!("Can't shutdown dispatcher thread: {:?}", e);
     }
+
+    // Be sure to call this _after_ draining the dispatcher
+    crate::launch_with_glean(|glean| {
```

Comment:
`launch_with_glean` uses the dispatcher. By the time this is called the dispatcher is shutdown and the code won't run.
I might have run over by [implementing a test for this](https://github.com/mozilla/glean/commit/300c286c8deb8e4e9ea4e010cf4fd1f3ace6d693).
It [fails on the current codebase](https://app.circleci.com/pipelines/github/mozilla/glean/8337/workflows/73efaadf-3f11-4a0b-a668-1736c3cc7cfb/jobs/150554) and does fail on this PR as is, it succeeds after fixing this.
(I think it would make sense to pull in the test).

```suggestion
    crate::with_glean(|glean| {
```

Reply:
Ah, of course. I knew that, of course I knew that. :facepalm: 

As for the test, we can probably write this as an integration test in Rust. Lemme try.

*some time later, after having re-remembered `-- --nocapture`*

Yup, it works.